order.go

```diff
@@ -0,0 +1,108 @@
+package apiutil
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+var (
+	// OrderASC defines the ascending order keyword
+	OrderASC = "asc"
+	// OrderDESC defines the descending order keyword
+	OrderDESC = "desc"
+)
+
+// OrderBy represents ordering information
+type OrderBy struct {
+	OrderItems []Order
+}
+
+// Order represents ordering information for a single key
+type Order struct {
+	Key       string
+	Direction string
+}
+
+// NewOrderBy constructs a basic order by struct
+func NewOrderBy() OrderBy {
+	return OrderBy{
+		OrderItems: []Order{},
+	}
+}
+
+// NewOrder constructs a basic order struct
+func NewOrder(key string, orderDirection string) Order {
+	if orderDirection != OrderASC && orderDirection != OrderDESC {
+		orderDirection = OrderDESC
+	}
+	return Order{
+		Key:       key,
+		Direction: orderDirection,
+	}
+}
+
+// AddOrdering allows to add items to the current ordering
+func (o *OrderBy) AddOrdering(order ...Order) {
+	o.OrderItems = append(o.OrderItems, order...)
+}
+
+// String implements the stringer interface for OrderBy
+func (o OrderBy) String() string {
+	var b bytes.Buffer
+	for i, oi := range o.OrderItems {
+		if i > 0 {
+			b.WriteString(",")
+		}
+		b.WriteString(oi.String())
+	}
+	return b.String()
+}
+
+// String implements the stringer interface for Order
+func (o Order) String() string {
+	return fmt.Sprintf("%s,%s", o.Key, o.Direction)
+}
+
+// SetQueryParams adds order by info to a url as query parameters
+func (o OrderBy) SetQueryParams(u *url.URL) *url.URL {
+	q := u.Query()
+	if len(o.OrderItems) > 0 {
+		q.Set("orderBy", o.String())
+	}
+
+	u.RawQuery = q.Encode()
+	return u
+}
+
+// OrderByFromRequest extracts orderBy params from an http request
+func OrderByFromRequest(r *http.Request) OrderBy {
+	orderBy := r.FormValue("orderBy")
+	return NewOrderByFromString(orderBy)
+}
+
+// NewOrderByFromString converts a commaa delimited string to an OrderBy struct
+func NewOrderByFromString(orderBy string) OrderBy {
```

Comment:
wondering if this should return an error on bad parsing. I'm sure you considered it. Thoughts?

Reply:
Compared to tha Page implementation, which has "defaults" passed in. Maybe we could do the same here?